pkg/util/log/zapcore_test.go

```diff
@@ -0,0 +1,128 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2021-present Datadog, Inc.
+
+package log
+
+import (
+	"bufio"
+	"bytes"
+	"fmt"
+	"strings"
+	"testing"
+
+	"github.com/cihub/seelog"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	"go.uber.org/zap"
+)
+
+func TestZapBasicLogging(t *testing.T) {
+	logger := NewZapLogger()
+	tests := []struct {
+		desc    string
+		f       func(*zap.Logger)
+		level   string
+		pattern string
+	}{
+		{
+			desc:    "Debug (no fields, debug level)",
+			f:       func(l *zap.Logger) { l.Debug("Simple message") },
+			level:   "debug",
+			pattern: "\\[DEBUG\\] Write: \\(log/zapcore_test.go:\\d+\\) \\| Simple message",
```

Comment:
I am not super happy with this way of writing the tests; it works but it's fragile: e.g. if you forget to escape a `|` the tests can easily pass on incorrect situations. I could just hardcode the line numbers which would be safer but more cumbersome to change.